src/index.js

```diff
@@ -9,11 +9,17 @@ function TestHook({ callback, hookProps, children }) {
 function renderHook(callback, { initialProps, ...options } = {}) {
   const result = { current: null }
   const hookProps = { current: initialProps }
+  const resolvers = []
+  const nextUpdate = () =>
+    new Promise((resolve) => {
+      resolvers.push(resolve)
+    })
 
   const toRender = () => (
     <TestHook callback={callback} hookProps={hookProps.current}>
       {(res) => {
         result.current = res
+        resolvers.splice(0, resolvers.length).forEach((resolve) => resolve())
```

Comment:
What if we pass res to resolve? Like this `resolvers.splice(0, resolvers.length).forEach((resolve) => resolve(res))`
Would it allow us to rename nextUpdate to nextValue and do assertions directly on it?
Like `expect(async nextValue).toEqual('Bob')`